app/models/payment.rb

```diff
@@ -44,42 +48,56 @@ def attrs
   end
 
   class BraintreeTransactionBuilder
+    #
+    # Stores and associates a Braintree transaction as +Payment::BraintreeTransaction+. Builder will also
+    # create an instance of +Payment::BraintreeCustomer+, if it doesn't already exist.
+    #
+    # === Options
+    #
+    # * +:action+                 - The ActiveRecord model of the corresponding action.
+    # * +:transaction_response+   - An Braintree::Transaction response object (see https://developers.braintreepayments.com/reference/response/transaction/ruby)
+    #
 
-    def self.build(page, transaction)
-      new(page, transaction).build
+    def self.build(action, transaction_response)
+      new(action, transaction_response).build
     end
 
-    def initialize(page, transaction)
-      @page = page
-      @transaction = transaction
+    def initialize(action, transaction_response)
+      @action = action
+      @transaction_response = transaction_response
     end
 
+
     def build
-      if @transaction.success?
+      if @transaction_response.success?
         ::Payment::BraintreeTransaction.create(transaction_attrs)
 
-
-        unless customer
-          ::Payment::BraintreeCustomer.create(customer_attrs)
+        unless locally_stored_customer
+          store_braintree_customer_locally
         end
       end
     end
 
     private
 
-    def customer
-      @customer ||= Payment.customer(customer_details.email)
+    def locally_stored_customer
+      @locally_stored_customer ||= Payment.customer(customer_details.email)
+    end
+
+    def store_braintree_customer_locally
+      Payment::BraintreeCustomer.create(customer_attrs)
     end
 
     def transaction_attrs
      {
-        transaction_id:         sale.id,
-        transaction_type:       sale.type,
-        amount:                 sale.amount,
-        transaction_created_at: sale.created_at,
-        merchant_account_id:    sale.merchant_account_id,
-        currency:               sale.currency_iso_code,
-        page:                   @page
+        transaction_id:          sale.id,
+        transaction_type:        sale.type,
+        payment_instrument_type: sale.payment_instrument_type,
```

Comment:
yes - (cc @eric) here's the acceptance test - https://github.com/SumOfUs/Champaign/pull/282/files#diff-5eaa2ff6985af0d0cc0ed9e93dbc8e22R82)